Add release-danube changelog

karimnaaji opened the pull request.

chloekraw commented on CHANGELOG.md:
```diff
@@ -1,3 +1,45 @@
+
+### 🍏 Improvements
+
+* An outstanding list of gesture issues and improvements were addressed in this release ([#9365](https://github.com/mapbox/mapbox-gl-js/pull/9365))
```
nit: HMmmm which meaning of "outstanding" are we trying to convey? Its presence strikes me as slightly off

chloekraw replied:
Also: do we think it's necessary to have this much detail in the changelog, especially when our changes are in accordance with general widely-adopted best practices for gestures? I lean towards a higher level summary and then a link to the PR for the rest.

karimnaaji replied:
Removed and focused on smaller set instead https://github.com/mapbox/mapbox-gl-js/pull/9542/commits/f5aa3f619aa1baa99f4a5ade94f28a8b3ec7e1d5.

@ansis Please confirm that this reflects your work appropriately.

chloekraw commented on CHANGELOG.md:
```diff
@@ -1,3 +1,45 @@
+  - _Changes to undocumented behavior_
+     - The previous gestures implementation prevented a camera change from actually happening if the interaction handler responsible for the change was disabled within the camera change event (`dragstart` or `rotatestart`). This behavior was not documented. It is no longer possible to prevent a camera change after it's event has been fired. It is possible some applications may be depending on thi ...
+     - The camera change events fired by the doubleClickZoom handler (`zoomstart`, `zoom`, `zoomend`, `movestart`, `move`, `moveend`) used to have undocumented properties such as `lngLat` and `point`. These have been removed
+* More precise collision detection for line labels ([#9219](https://github.com/mapbox/mapbox-gl-js/pull/9219))
```
I appreciate the greater clarity that comes from breaking out this PR into sub-bullets, but given how long this changelog already is from the gestures improvement, I wonder whether we can shorten the syntax of some of the improvements and keep things as a single entry, e.g.:

> Line labels now have improved collision detection, with greater precision in placement, reduced memory footprint, better placement under pitched camera orientations.

karimnaaji replied:
Updated in https://github.com/mapbox/mapbox-gl-js/pull/9542/commits/a1219354822fe417ce65bf1abf8377966f6a41a3

chloekraw commented on CHANGELOG.md:
```diff
@@ -1,3 +1,45 @@
+  - More tightly packed labels during placement, resulting in more visible labels
+  - Reduced memory footprint
+* Fix `GlyphManager` continually re-requesting missing glyphs ranges ([#8027](https://github.com/mapbox/mapbox-gl-js/issues/8027), fixed by [#9375](https://github.com/mapbox/mapbox-gl-js/pull/9375)) (h/t [oterral](https://github.com/oterral))
+* Remove Hershey Simplex font from our licence, which no longer needs attribution ([#9392](https://github.com/mapbox/mapbox-gl-js/issues/9392))
```
I'm not sure I think that this needs to be our changelog. 

If we decide to include it, we should explain why it no longer needs attribution. I assume it's because we're no longer using it, in which case we could write: 

> Remove attribution for the Hershey Simplex font from license.txt, which this project no longer uses.

chloekraw replied:
Also, we should stick to the spelling of license that matches the name of the file

karimnaaji replied:
I was not sure about this. Removed in 1f48c6396dbfc9caae467398fd73f8e779b2653f.

chloekraw commented on CHANGELOG.md:
```diff
@@ -1,3 +1,45 @@
+  - Improves placement under various camera orientation
+  - More tightly packed labels during placement, resulting in more visible labels
+  - Reduced memory footprint
+* Fix `GlyphManager` continually re-requesting missing glyphs ranges ([#8027](https://github.com/mapbox/mapbox-gl-js/issues/8027), fixed by [#9375](https://github.com/mapbox/mapbox-gl-js/pull/9375)) (h/t [oterral](https://github.com/oterral))
```
Oh nice! I'm glad this got fixed. Should it be `glyph ranges` instead of glyphs?

chloekraw commented on CHANGELOG.md:
```diff
@@ -1,3 +1,45 @@
+* Add `index-of` and `slice` expressions to search arrays and strings for the first occurrence of a specified value and return a section of the original array or string ([#9450](https://github.com/mapbox/mapbox-gl-js/pull/9450)) (h/t [lbutler](https://github.com/lbutler))
+* Correctly set RTL text plugin status if the plugin URL could not be loaded. This allows developers to add retry logic on network errors when loading the plugin ([#9489](https://github.com/mapbox/mapbox-gl-js/pull/9489))
+
+### 🍏 Improvements
```
I like the practice we adopt on mobile of being flexible with the sub-heading names when a release features a particularly large number of changes centered on a specific category. For example, see how the iOS team handled the recent snapshotter refactor: https://github.com/mapbox/mapbox-gl-native-ios/releases/tag/ios-v5.8.0 

For this release, the changelog could be organized as:
```
### Features

### Gestures

### Other improvements
```

karimnaaji replied:
Updated

karimnaaji commented:
@chloekraw Thanks for the review!

chloekraw commented on CHANGELOG.md:
```diff
@@ -1,3 +1,39 @@
+* Add `index-of` and `slice` expressions to search arrays and strings for the first occurrence of a specified value and return a section of the original array or string ([#9450](https://github.com/mapbox/mapbox-gl-js/pull/9450)) (h/t [lbutler](https://github.com/lbutler))
+* Correctly set RTL text plugin status if the plugin URL could not be loaded. This allows developers to add retry logic on network errors when loading the plugin ([#9489](https://github.com/mapbox/mapbox-gl-js/pull/9489))
+
+### Gestures ([#9365](https://github.com/mapbox/mapbox-gl-js/pull/9365))
```
```suggestion
### Gestures 
This release significantly refactors and improves gesture handling on desktop and mobile. Three new touch gestures have been added: `two-finger swipe` to adjust pitch, `two-finger double tap` to zoom out, and `tap then drag` to adjust zoom with one finger ([#9365](https://github.com/mapbox/mapbox-gl-js/pull/9365)). In addition, this release brings the following changes and bug fixes:
```

chloekraw replied:
@karimnaaji @ansis an extra set of eyes here to ensure I’ve described each gesture correctly would be good

chloekraw commented on CHANGELOG.md:
```diff
@@ -1,3 +1,39 @@
+- You can now interact with multiple maps on the same page at the same page. For example, pinch rotate one map while touch panning a different one
+add a second finger, keep panning, remove the first, keep panning
+- Fix map jump when releasing one finger after pinch zoom ([#9136](https://github.com/mapbox/mapbox-gl-js/issues/9136))
+- Stop mousedown and touchstart from interrupting easeTo animations if interaction handlers are disabled ([#8725](https://github.com/mapbox/mapbox-gl-js/issues/8725))
```
```suggestion
- Stop mousedown and touchstart from interrupting `easeTo` animations when interaction handlers are disabled ([#8725](https://github.com/mapbox/mapbox-gl-js/issues/8725))
```

chloekraw commented on CHANGELOG.md:
```diff
@@ -1,3 +1,39 @@
+- Fix map jump when releasing one finger after pinch zoom ([#9136](https://github.com/mapbox/mapbox-gl-js/issues/9136))
+- Stop mousedown and touchstart from interrupting easeTo animations if interaction handlers are disabled ([#8725](https://github.com/mapbox/mapbox-gl-js/issues/8725))
+- Stop mouse wheel from interrupting animations when `map.scrollZoom` is disabled ([#9230](https://github.com/mapbox/mapbox-gl-js/issues/9230))
+- The previous gestures implementation prevented a camera change from actually happening if the interaction handler responsible for the change was disabled within the camera change event (`dragstart` or `rotatestart`). This behavior was not documented. It is no longer possible to prevent a camera change after it's event has been fired. It is possible some applications may be depending on this beh ...
```
```suggestion
- A camera change can no longer be prevented by disabling the interaction handler within the camera change event. Selectively prevent camera changes by listening to the `mousedown` or `touchstart` map event and then calling [.preventDefault()](https://docs.mapbox.com/mapbox-gl-js/api/#mapmouseevent#preventdefault) ([#9365](https://github.com/mapbox/mapbox-gl-js/pull/9365))
```

chloekraw commented on CHANGELOG.md:
```diff
@@ -1,3 +1,39 @@
+- Stop mousedown and touchstart from interrupting easeTo animations if interaction handlers are disabled ([#8725](https://github.com/mapbox/mapbox-gl-js/issues/8725))
+- Stop mouse wheel from interrupting animations when `map.scrollZoom` is disabled ([#9230](https://github.com/mapbox/mapbox-gl-js/issues/9230))
+- The previous gestures implementation prevented a camera change from actually happening if the interaction handler responsible for the change was disabled within the camera change event (`dragstart` or `rotatestart`). This behavior was not documented. It is no longer possible to prevent a camera change after it's event has been fired. It is possible some applications may be depending on this beh ...
+- The camera change events fired by the doubleClickZoom handler (`zoomstart`, `zoom`, `zoomend`, `movestart`, `move`, `moveend`) used to have undocumented properties such as `lngLat` and `point`. These have been removed
```
```suggestion
- Undocumented properties on the camera change events fired by the doubleClickZoom handler (`zoomstart`, `zoom`, `zoomend`, `movestart`, `move`, `moveend`) have been removed ([#9365](https://github.com/mapbox/mapbox-gl-js/pull/9365))
```

chloekraw commented on CHANGELOG.md:
```diff
@@ -1,3 +1,39 @@
+
+### Gestures ([#9365](https://github.com/mapbox/mapbox-gl-js/pull/9365))
+
+- You can now interact with multiple maps on the same page at the same page. For example, pinch rotate one map while touch panning a different one
+add a second finger, keep panning, remove the first, keep panning
```
🤔 I'm quite confused by this example: `For example, pinch rotate one map while touch panning a different one
add a second finger, keep panning, remove the first, keep panning` -- was something accidentally deleted from L12?

In any case, I think "You can now interact with multiple maps on the same page at the same page." is sufficiently clear without needing an example -- people can head to the PR for details,  so I recommend removing.

chloekraw replied:
```suggestion
- It's now possible to interact with multiple maps on the same page at the same time [#9365](https://github.com/mapbox/mapbox-gl-js/pull/9365)
```

chloekraw commented on CHANGELOG.md:
nit: **possible** undefined exception

chloekraw commented on CHANGELOG.md:
nit: Fix a bug where the `within` expression would incorrectly evaluate geometries that cross the antimeridian

chloekraw commented on CHANGELOG.md:
Typo: values 

Or:

> Fix a bug where the `within` expression would inconsistently evaluate points on tile boundaries

chloekraw commented on CHANGELOG.md:
nit: Dependency upgrades are usually in an “Other changes” section. We can keep it here as the last bullet alternatively.

karimnaaji replied:
I left it since it would be quite a lonely section for this time. But reordered it last.

chloekraw commented on CHANGELOG.md:
nit: `background-pattern`

karimnaaji commented:
> Improvements and bug fixes
>I don’t feel strongly about this but I think it looks cleaner
@chloekraw Agreed, updated for that and addressed the other nits
